src/main/java/htsjdk/tribble/index/IndexFactory.java

```diff
@@ -174,18 +174,30 @@ public static Index loadIndex(final String indexFile) {
      *                     {@link java.nio.file.Path}
      */
     public static Index loadIndex(final String indexFile, Function<SeekableByteChannel, SeekableByteChannel> indexWrapper) {
+        try {
+            return loadIndex(indexFile, indexFileInputStream(indexFile, indexWrapper));
+        } catch (final IOException ex) {
+            throw new TribbleException.UnableToReadIndexFile("Unable to read index file", indexFile, ex);
+        }
+    }
+
+    /**
+     * Load in index from the specified stream. Note that the caller is responsible for decompressing the stream if necessary.
+     *
+     * @param source the stream source (typically the file name)
+     * @param inputStream the raw byte stream of the index
+     */
+    public static Index loadIndex(final String source, final InputStream inputStream) {
```

Comment:
@tomwhite Could you add an explicit test for this new method?  It can reuse the existing test code for the previous methods if you want.

Reply:
Added a simple test.